DEVELOPING.md

```diff
@@ -0,0 +1,51 @@
+OpenSim is a community resource that is housed in the OpenSim-Core repository.
+
+This specific document addresses issues that come up during development and maintenance of OpenSim code base, that may not be relevant to end users because it's not in release code yet or of histroical relevance only but would be 
+valuable to new developers or advanced users. For general contribution guidelines please consult [CONTRIBUTING.md](https://github.com/opensim-org/opensim-core/blob/master/CONTRIBUTING.md)
+
+Contents:
+
+- [Backward Compatibility](#backward-compatibility)
+
+
+Backward Compatibility
+
+OpenSim stores models and analysis-tools/objects in xml files (with extension .osim or .xml). Since first release of OpenSim in 2007, these files have mostly survived many major format changes which is quite important considering that thousands of these files are around both in earlier OpenSim distributions and also in models created by users and/or collaborators. This section describes how backward compatibility is maintained so that if a format change is introduced, users do not lose valuable work and/or models in the process. In what follows we use the term serialization to mean writing and deserialization to mean reading. 
+
+One key piece of information that is kept with each file is a version number (usually in the header section e.g. ``` <OpenSimDocument Version="NNNNN">```. Same is true for .sto files that contain a header as well though will not discuss here. The version number encodes what the code was expecting the format to be when the file was produced/written. Keeping this number consistent with the content of the file is the fundamental invariant that needs to be maintained. In code, the version number is maintained in XMLDocument.cpp and is updated as version numbers are incremented. The actual numbers do not matter much as long as they are monotonically increasing. OpenSim 3.2 shipped with OpenSimDocument Version number 30000. OpenSim 4.0 development started at version 30500 and will ship with 40000).
+
+Deserialization of Objects and their Properties (primitives that are read-from/written-to files)  is mostly transparent to users/developers. Once you use the property macros (e.g. `` OpenSim_DECLARE_UNNAMED_PROPERTY```) in the header and perform the necessary wiring/calls, these properties know how to serialize themselves to XML elements and back. This serialization/deserialization, however, assumes a fixed XML layout/schema. If this assumption fails (because of property name change, layout change or addition of new properties) then deserialization will not work out of the box and the developer making the change is responsible for changing the version number and the deserialization code to be in sync. with the latest code. This is very important since if the developer does not update the version number then the code and the XML files will not be in sync and it will be very hard to go back to find out what changes should have been made. 
```

Comment:
Just use single back-quotes in line -- the triple back quotes are just for blocks of code. And here there are two on one side and three on the other.